src/mui/detail/Edit.js

```diff
@@ -72,25 +70,18 @@ export class Edit extends Component {
             data,
         });
         const titleElement = data ? <Title title={title} record={data} defaultTitle={defaultTitle} /> : '';
-        // using this.previousKey instead of this.fullRefresh makes
-        // the new form mount, the old form unmount, and the new form update appear in the same frame
-        // so the form doesn't disappear while refreshing
-        const isRefreshing = key !== this.previousKey;
-        this.previousKey = key;
-
         return (
             <div className="edit-page">
-                <Card style={{ opacity: isLoading ? 0.8 : 1 }} key={key}>
```

Comment:
I think the objective is to force a refetch in child components. For instance, if the `Edit` view contains a `ReferenceInput`, and that the list of possible values for that field has changed server-side, a hit on "Refresh" should force the `ReferenceInput` to remount to fetch the data. That's what the `key={version}` does.